File: src/BinarySearch.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class BinarySearch {
    private static long[] arrayOfLong;

    public static void main(String[] args) throws FileNotFoundException {
        run();
    }

    private static void run() throws FileNotFoundException {
        Scanner scanner = new Scanner(new File("C:\\Devel\\Algorithms-theory-and-practice\\src\\input.txt"));
        int n = scanner.nextInt();
        arrayOfLong = new long[n];
        for (int i = 0; i < n; i++) {
            arrayOfLong[i] = scanner.nextLong();
        }
        int k = scanner.nextInt();
        for (int i = 0; i < k; i++) {
            int index = binarySearch(scanner.nextLong());
            System.out.print(String.valueOf(index) + " ");
        }
    }

    private static int binarySearch(long seek) {
        int left = 0;
        int right = arrayOfLong.length - 1;
        while (left <= right) {
            int middle = (left + right) / 2;
            if (arrayOfLong[middle] == seek) {
                return middle + 1;
            } else if (arrayOfLong[middle] > seek) {
                right = middle - 1;
            } else {
                left = middle + 1;
            }
        }
        return -1;
    }
}

File: src/binary_search.py
def binary_search(array, right_border, seek):
    left, right = 0, right_border
    while left <= right:
        middle = (left + right) // 2
        if array[middle] == seek:
            return middle + 1
        elif array[middle] > seek:
            right = middle - 1
        else:
            left = middle + 1
    return -1


def main():
    n, *data = map(int, input().split())
    _, *seeks = map(int, input().split())
    for seek in seeks:
        print(binary_search(data, n - 1, seek), end=' ')


if __name__ == '__main__':
    main()
